Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -478,48 +589,81 @@ bool AzAssetBrowserRequestHandler::CanAcceptDragAndDropEvent(
         return false;
     }
 
-    bool canAcceptEvent = false;
+    return DecodeDragMimeData(event->mimeData());
+}
 
-    // Detects Source Asset Entries whose extensions are handled by a system
-    AzToolsFramework::AssetBrowser::AssetBrowserEntry::ForEachEntryInMimeData<SourceAssetBrowserEntry>(
-        event->mimeData(), [&](const AzToolsFramework::AssetBrowser::SourceAssetBrowserEntry* source) {
-            if (AssetBrowser::AssetBrowserSourceDropBus::HasHandlers(source->GetExtension()))
-            {
-                if (outSources.has_value())
-                {
-                    outSources.value()->push_back(source);
-                }
-                canAcceptEvent = true;
-            }
-        });
+bool AzAssetBrowserRequestHandler::DecodeDragMimeData(const QMimeData* mimeData,
+    AZStd::vector<const AzToolsFramework::AssetBrowser::ProductAssetBrowserEntry*>* outProducts) const
+{
+    using namespace AzToolsFramework;
+    using namespace AzToolsFramework::AssetBrowser;
+    using namespace AzAssetBrowserRequestHandlerPrivate;
 
-    // Detects Product Assets that are dragged directly, or child Products of other entry types.
-    AzToolsFramework::AssetBrowser::AssetBrowserEntry::ForEachEntryInMimeData<ProductAssetBrowserEntry>(
-        event->mimeData(), [&](const AzToolsFramework::AssetBrowser::ProductAssetBrowserEntry* product)
+    // what we'd like to do with drop events is spawn an entity per selected logical asset
```

Comment:
nit: I suggest that all references to 'spawn', 'spawnList', etc in the PR are reworded to either 'create' or 'instantiate'